pkg/agent/flexvolume/controller.go

```diff
@@ -116,12 +116,18 @@ func (c *FlexvolumeController) Attach(attachOpts AttachOptions, devicePath *stri
 			if index != -1 {
 				// check if the RW attachment is orphaned.
 				attachment := &volumeattachObj.Attachments[index]
+
+				logger.Infof("Volume attachment record %s/%s exists for pod: %s/%s", volumeattachObj.Namespace, volumeattachObj.Name, attachment.PodNamespace, attachment.PodName)
+				// Note this could return the reference of the pod who is requesting the attach if this pod have the same name as the pod in the attachment record.
 				pod, err := c.context.Clientset.Core().Pods(attachment.PodNamespace).Get(attachment.PodName, metav1.GetOptions{})
-				if err != nil {
-					if !errors.IsNotFound(err) {
+				if err != nil || (attachment.PodNamespace == attachOpts.PodNamespace && attachment.PodName == attachOpts.Pod) {
```

Comment:
in the case where there is an existing attachment record for a pod of the same name we are trying to attach to now, what is the status of that pod?  would it be pending/creating because the new pod "exists" but hasn't started yet because we haven't finished attaching?

Reply:
if the attachment record shows a pod that has the same name and namespace as the one we are trying to attach, then it is the same pod. Otherwise kubernetes api would've error out since you cant have two resources of the same name in the same namespace. So the status will always be pending, since it is refering to itself and it is in a chicken-egg state.

This would've been better fix if there was a way to get pods via UID as well as the name. but i can see how this could be confusing.